pkg/agent/util/iptables/iptables.go

```diff
@@ -75,6 +75,26 @@ const (
 // iptables-restore: support acquiring the lock.
 var restoreWaitSupportedMinVersion = semver.Version{Major: 1, Minor: 6, Patch: 2}
 
+type Interface interface {
+	EnsureChain(protocol Protocol, table string, chain string) error
+
+	ChainExists(protocol Protocol, table string, chain string) (bool, error)
+
+	AppendRule(protocol Protocol, table string, chain string, ruleSpec []string) error
+
+	InsertRule(protocol Protocol, table string, chain string, ruleSpec []string) error
+
+	DeleteRule(protocol Protocol, table string, chain string, ruleSpec []string) error
+
+	DeleteChain(protocol Protocol, table string, chain string) error
+
+	ListRules(table string, chain string) ([]string, error)
+
+	Restore(data string, flush bool, useIPv6 bool) error
+
+	Save() ([]byte, error)
```

Comment:
can you explain why `Save` returns a `[]byte` but `Restore` takes a `string` as parameter? Intuitively the 2 functions should be "symmetric".

Reply:
For `Restore`, if we use `[]byte` as the argument type, it's difficult to know what's the actual argument the method was called with when the test fails. The error message would show two long slices of bytes which are not readable, and developer needs to change the argument type like I do in this patch to get what's wrong.
For `Save`, we could also change it to return string, but the above readability issue doesn't exist as we could convert the returned `[]byte` to string in test code. Since `exec.Command`returns `[]byte` and the only consumer `dumpIPTables` uses `[]byte`, I didn't make the change to avoid two conversions.

Reply:
Let me know if you think making them consistent is necessary.